rules/legacy_xcodeproj.bzl

```diff
@@ -1058,6 +1087,7 @@ def _xcodeproj_impl(ctx):
         "BAZEL_CONFIGS": ctx.attr.configs.keys(),
         "BAZEL_ADDITIONAL_BAZEL_BUILD_OPTIONS": " ".join(["{} ".format(opt) for opt in ctx.attr.additional_bazel_build_options]),
         "BAZEL_ADDITIONAL_LLDB_SETTINGS": "\n".join(ctx.attr.additional_lldb_settings),
+        "INDEX_DATA_STORE_DIR": "$(INDEX_DATA_STORE_DIR)",
```

Comment:
So this is set by xcconfig but no longer passed to the script? This seems fine

Reply:
This is set here because otherwise `INDEX_DATA_STORE_DIR` is empty during scripts phase. Tested this in both Xcode 13.x and 14.x and when this setting is set `INDEX_DATA_STORE_DIR` contains the respective path to the index store. Most likely Xcode needs this value explicitly requested in a build setting to make it non-empty during script phase.